library/src/main/java/org/xmtp/android/library/Conversation.kt

```diff
@@ -28,154 +29,130 @@ sealed class Conversation {
-            is Group -> group.consentState()
+            is V1 -> throw XMTPException("Only supported for V3")
+            is V2 -> throw XMTPException("Only supported for V3")
+            is Group -> group.isCreator()
+            is Dm -> dm.isCreator()
         }
     }
 
-    /**
-     * This method is to create a TopicData object
-     * @return [TopicData] that contains all the information about the Topic, the conversation
-     * context and the necessary encryption data for it.
-     */
-    fun toTopicData(): TopicData {
-        val data = TopicData.newBuilder()
-            .setCreatedNs(createdAt.time * 1_000_000)
-            .setPeerAddress(peerAddress)
+    suspend fun members(): List<Member> {
         return when (this) {
-            is V1 -> data.build()
-            is V2 -> data.setInvitation(
-                Invitation.InvitationV1.newBuilder()
-                    .setTopic(topic)
-                    .setContext(conversationV2.context)
-                    .setAes256GcmHkdfSha256(
-                        Aes256gcmHkdfsha256.newBuilder()
-                            .setKeyMaterial(conversationV2.keyMaterial.toByteString()),
-                    ),
-            ).build()
-
-            is Group -> throw XMTPException("Groups do not support topics")
+            is V1 -> throw XMTPException("Only supported for V3")
+            is V2 -> throw XMTPException("Only supported for V3")
+            is Group -> group.members()
+            is Dm -> dm.members()
         }
     }
 
-    fun decode(envelope: Envelope, message: MessageV3? = null): DecodedMessage {
+    suspend fun updateConsentState(state: ConsentState) {
         return when (this) {
-            is V1 -> conversationV1.decode(envelope)
-            is V2 -> conversationV2.decodeEnvelope(envelope)
-            is Group -> message?.decode() ?: throw XMTPException("Groups require message be passed")
-        }
-    }
-
-    fun decodeOrNull(envelope: Envelope): DecodedMessage? {
-        return try {
-            decode(envelope)
-        } catch (e: Exception) {
-            Log.d("CONVERSATION", "discarding message that failed to decode", e)
-            null
-        }
-    }
-
-    fun <T> prepareMessage(content: T, options: SendOptions? = null): PreparedMessage {
-        return when (this) {
-            is V1 -> {
-                conversationV1.prepareMessage(content = content, options = options)
-            }
-
-            is V2 -> {
-                conversationV2.prepareMessage(content = content, options = options)
-            }
-
-            is Group -> throw XMTPException("Groups do not support prepared messages") // We return a encoded content not a preparedmessage which requires a envelope
+            is V1 -> throw XMTPException("Only supported for V3")
+            is V2 -> throw XMTPException("Only supported for V3")
+            is Group -> group.updateConsentState(state)
+            is Dm -> dm.updateConsentState(state)
```

Comment:
It's implemented on the client in a method on contacts. When it was originally written it was just consenting on addresses so that made sense. We could probably update the state here for V1 and V2 but feels unnecessary as we don't want people building new V2 functionality and instead moving to V3. https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/Contacts.kt